### PR TITLE
[Semi-modular] Gives Xeno-Hybrids a plasmavessel, resinspinner and hivenode at the cost of high fire weakness

### PR DESCRIPTION
**Pull request by desertmote**
## About The Pull Request

This PR adds a `/roundstart` plasmavessel to Xeno-hybrids, it caps out at 55 plasma which is what it takes to create 1 resin structure. It regenerates at a low 2 plasma per tick.
It gives the Xeno-hybrids the resinspinner, which allows them to create resin structures at the cost of plasma.
It also gives them hivenode, which allows telephatic-esque whispering at the cost of plasma.

The trade-off for being able to create structures from a regenerating resource is that that xeno-hybrids are INSANELY weak to fire now. As demonstrated in the video below you'll be able to tell that the xeno-hybrid hardcrits way earlier than a regular anthromorph.

## How This Contributes To The Skyrat Roleplay Experience

We have unique species that deserve to have some of their flavor, this encourages more players to play xeno-hybrid. I also genuinely think this can lead to good roleplay scenarios where a xeno-hybrid may cause a **healthy conflict** with their resin, which allows the rest of the department/security to react appropriately.

Because we do not run actual xenomorphs, and likely wont in the foreseeable future as both us and TG hardly invest maintaining time into it, I think its a great idea to put some of their tools into our hybrid species.

## Proof of Testing

https://github.com/Skyrat-SS13/Skyrat-tg/assets/77534246/caf6efa1-f527-460a-b315-9ba960f4d10f

## Changelog

:cl:
add: Xeno-hybrids have a weak plasmavessel to spin resin and hive communicate with
add: Xeno-hybrids have acid blood
balance: Xeno-hybrids are the weakest species against heat/fire, at a whopping 2.5x multiplier
balance: Xeno-hybrid livers can process plasma without gaining toxin damage
/:cl:

### CODE REVIEW

**Comment by Nerev4r** [comment]
2 plasma a tick is pretty low for the 55 structure cost, but it's pretty balanced at the same time. I like this.

**Comment by desertmote** [comment]
> 2 plasma a tick is pretty low for the 55 structure cost, but it's pretty balanced at the same time. I like this.

You can see its recharge speed in the video when I open the VV panel, I may have to lower it even more because I think its still rather quick.

**Comment by Paxilmaniac** [comment]
Based

**Comment by ghost** [comment]
They really shouldn't be able to place the resin floors that spread. It's just going to be seen as a griefing tool to be able to cover a room in resin flooring.

**Comment by desertmote** [comment]
> They really shouldn't be able to place the resin floors that spread. It's just going to be seen as a griefing tool to be able to cover a room in resin flooring.

Its simply not. The floor beneath the resin is unchanged, the resin is removed by simple tools, it is placable at a once-per-minute intervals.
What makes you think its griefing? If a player is placing them "too much", why is it rulebreak and not an IC issue?

Really curious what your perspective here is, because to me it makes very little sense at all to call this a grief tool when any actual construction or artistic tools ingame do worse "damage".

**Comment by ghost** [comment]
> Really curious what your perspective here is, because to me it makes very little sense at all to call this a grief tool when any actual construction or artistic tools ingame do worse "damage".

image
I'm pointing out that it's possibly going to cause issues. Not saying it IS anything.

Either way, though:
https://github.com/Skyrat-SS13/Skyrat-tg/pull/9704
image

It was already fairly unanimous that xeno resin shouldn't be on-station unless there's an actual xeno threat. Granted, the linked PR wasn't about specifically hybrids, the responses mention that they're already toeing a line.

**Comment by desertmote** [comment]
That PR was to give non-xeno-hybrids the option to get the resin emote, which xeno-hybrids already have currently. Its not relevant to this xeno-hybrid only PR.
It is also from _two years ago_, during a drastically different server climate that was running into real xenomorphs near weekly.

**Comment by Nerev4r** [comment]
> > Really curious what your perspective here is, because to me it makes very little sense at all to call this a grief tool when any actual construction or artistic tools ingame do worse "damage".
> 
> image I'm pointing out that it's possibly going to cause issues. Not saying it IS anything.
> 
> Either way, though: #9704 image
> 
> It was already fairly unanimous that xeno resin shouldn't be on-station unless there's an actual xeno threat. Granted, the linked PR wasn't about specifically hybrids, the responses mention that they're already toeing a line.

image

the problem is we're pretty committed to not having xenos these days

**Comment by Lutowski** [comment]
Solid PR, xeno-hybrids deserve some love.

**Comment by Namwty** [comment]
Building raisin walls and such should have a build time of couple seconds, same reason as why most stuff have a build time ( genetic web production and generally walls, apart from some rare instances like... snowmen )

I doubt that anyone wants to get fortnited in maints and alike.

And 2.5 burn multiplier is nuts, no offense... even something like welder will hit em harder than a dblade. Not even going to talk about most laser weapons will instantly disable limbs and crit/kill within 2/3 shots ( depending on weapon ).
Yeah sure... lasers aren't used frequently in SR, but think of outer sources, people friendly firing during mold or just firing lasers in hallways while chasing someone, tesliums, hacked vendors, shocked doors, cold/hot from temperature and so on.
Snails have 2x and that shit is absurd as hell, armor or no armor.... anything above that is just borderline unplayable.

My point is some folk will just stop playing xeno-hybrid characters full stop if they basically get put into a coffin every time anything with burns happen or whenever someone they fight just takes out a welder, regardless of how fun and rp potential this plasmavessel stuff adds. I think there should be more reasonable way to balance it out, that is less frustrating.

EDIT: And I'm not even sure RP oriented mostly ability would require such harsh balancing. Other races get some RP related stuff as well combat stuff, and their balancing is laughable and sometimes even none.

**EDIT2: Nevermind, I'm a dumbass and I misread heat weakness as burn weakness... that changes it quite a lot and isn't that bad.**

**Comment by desertmote** [comment]
2.5x is probably fine, since synths are 2x already i cant lower it much, lest it would no longer be the weakest species against fire.

If you're afraid of being walled off in maints being too powerful, just don't fight a xenonybrid in maints. Its thematic, but most importantly a good combat diversity. After all, they and their resin are painfully weak to fire, so prepare well or don't chase into a bad situation.

Remember it takes like a minute+ for them to place resin structures, if you can't break it faster than its being built you're probably hitting it with your fists.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/xeno.dm [hunk 1]
```diff
@@ -11,6 +11,9 @@
 	)
 	inherent_biotypes = MOB_ORGANIC|MOB_HUMANOID
 	mutanttongue = /obj/item/organ/internal/tongue/xeno_hybrid
+	mutant_organs = list(/obj/item/organ/internal/alien/plasmavessel/roundstart, /obj/item/organ/internal/alien/resinspinner, /obj/item/organ/internal/alien/hivenode)
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/xeno.dm]:
```suggestion
	mutant_organs = list(
		/obj/item/organ/internal/alien/plasmavessel/roundstart,
		/obj/item/organ/internal/alien/resinspinner,
		/obj/item/organ/internal/alien/hivenode,
	)
```

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/xeno.dm [hunk 2]
```diff
@@ -11,6 +11,9 @@
...
+	coldmod = 0.8
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/xeno.dm]: This cold modifier wasn't mentioned anywhere.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/xeno.dm [hunk 3]
```diff
@@ -52,3 +74,8 @@
 	xeno.dna.mutant_bodyparts["xenohead"] = list(MUTANT_INDEX_NAME = "Standard", MUTANT_INDEX_COLOR_LIST = list(xeno_color, xeno_color, xeno_color))
 	regenerate_organs(xeno, src, visual_only = TRUE)
 	xeno.update_body(TRUE)
+
+/obj/item/organ/internal/alien/plasmavessel/roundstart
+	stored_plasma = 55
+	max_plasma = 55
+	plasma_rate = 2
```
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/xeno.dm:92]: This gives them a *very* fast heal rate when standing on resin floors, healing their brute, burn and oxygen damage at a by 0.2*2.5 = 0.5/s each.
For now, I'd rather not have to consider or worry about xeno-hybrids being able to heal using this,
 so just set the `heal_rate` to `0` for the time being.
```suggestion
	plasma_rate = 2
	heal_rate = 0
```

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/xeno.dm (hunk 1 above)
Comment (GoldenAlpharex) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/xeno.dm]: I'd also rather they have at least some `do_after()` for putting down resin, just so it's coherent with the rest of the building in the game currently. A simple `0.5 SECONDS` should do.
In order to do this, you would simply need to create a new variable on `/datum/action/cooldown/alien/make_structure/resin` (i.e. `building_duration`) that would be set to `0` and checked in its `Activate()` after the `owner.visible_message()` to see if it does a `do_after()` (i.e. `if(building_duration && !do_after(owner, building_duration))` with a message before a return to say that you stopped shaping the resin or whatever). Then you'd do a new subtype of that action (i.e. `/datum/action/cooldown/alien/make_structure/resin/roundstart`) where you'd simply set that new variable to `0.5 SECONDS`. Finally, you'd create a new subtype of the resin spinner (i.e. `/obj/item/organ/internal/alien/resinspinner/roundstart`), and assign its `actions_types` to a list containing only that new subtype of the action that you created earlier. That should be all that's needed for this to work.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I'm willing to give this a try, but only if these requirements are met. I'd rather not have this be make it a species of pseudo-antags, especially considering it's not veteran-locked.

**Comment by desertmote** [comment]
Tested and edited!

https://github.com/Skyrat-SS13/Skyrat-tg/assets/77534246/7a94b095-a204-431d-87ab-b48503a839cd

I've also made an addition, namely xenohybrids having acid blood (No gameplay changes besides what you should fill yourself up with), and xenohybrid livers being able to process plasma without gaining toxin damage.
I also gave some of the organs a different sprite for flavor.

**Comment by Gandalf2k15** [comment]
Given we're under new leadership, what about the fact this could make the crew feel even more acclimatised to seeing xeno floors and similar, meaning that if real xenos show up, they are less likely to think they are real when they see a resin floor?

**Pull request closed without merging**